### PR TITLE
send routes to nav native only when trip session is started

### LINKED ISSUE #4727: Native observer can miss events
We are subscribing the `NavigatorObserver` in the `MapboxTripSession` only when the `start()` function is called. This means that in situations where, for example, a route is set _before_ the session is started, all of the events produced by Nav Native while the route is set (including banner instructions updates) are going to be dropped by the Nav SDK.

We should look into refactoring `MapboxTripSession` to always register the `NavigatorObserver` when `MapboxTripSession` is initialized to make sure that we're not dropping any events, and make sure that the events are not propagated further if the session is stopped. Then, when a session is started, we should send all of the accumulated latest information (including observer calls).

cc @mapbox/navigation-android

### PR DESCRIPTION
**Pull request by Zayankovsky**
### Description
Closes #4727.

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt [hunk 1]
```diff
@@ -244,8 +244,6 @@ internal class MapboxTripSession(
                         ) {
                             // workaround for a remaining issues in
                             // github.com/mapbox/mapbox-navigation-native/issues/3466
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:246]: I'm still able to reproduce https://github.com/mapbox/mapbox-navigation-android/issues/4727#issuecomment-926810538

> Noting that I'm lately missing the very first voice instruction and it seems definitely related to OP.

@SiarheiFedartsou @mskurydin @DmitryAk could you explain why the very first status is generating `null` `bannerInstruction` and `voiceInstruction`?

first navigation status null banner and voice instructions

It seems that `#3466` is not fully solved.

cc @S2Ler

Comment (Zayankovsky) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:246]: If I'm not mistaken, for `#3466` we need to switch to `setRouteAsync` function, which is what @korshaknn is doing in #4881. So in order to fully resolve issues with banner/voice instructions, we need to merge both this fix and #4881.

Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:246]: That makes sense. Thanks for clarifying 🙇‍♂️

**Review by Guardiola31337 — APPROVED** [review]
LGTM

**Comment by Guardiola31337** [comment]
@Zayankovsky does this need back-porting? 🤔 

cc @LukasPaczos

**Comment by LukasPaczos** [comment]
I'd say that we shouldn't risk it for GA and rather let it curate in 2.1 for now.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt (hunk 1 above)
Comment (LukasPaczos) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:246]: We shouldn't really need async route setters to resolve missing voice instructions :thinking: 

If we are setting the route to NN only after the trip session starts, and all the observers are already subscribed when a  session starts, then we shouldn't miss the first voice event.

Comment (LukasPaczos) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:246]: Ah, I see you raised it in https://github.com/mapbox/mapbox-navigation-native/issues/3466#issuecomment-930539819 :+1:

**Comment by Guardiola31337** [comment]
> I'd say that we shouldn't risk it for GA and rather let it curate in 2.1 for now.

Discussed internally, as https://github.com/mapbox/mapbox-navigation-android/pull/4881 is introducing breaking changes and OP is referencing it, we're going to backport it to `2.0`. Adding the `needs-backporting` label.

**Pull request merged**